app/services/editions/publishing/PublicationTransform.scala

```diff
@@ -0,0 +1,12 @@
+package services.editions.publishing
+
+import model.editions.PublishableIssue
+import play.api.libs.json.Writes
+
+/**
+ * PublicationTransform is a generic interface that takes a PublishableIssue and turns it into
+ * some other shape, which must be JSON-serializable
+ */
+trait PublicationTransform[O] {
```

Comment:
Should this be a polymorphic trait – or, put another way, is this something the publication target hierarchy needs to know about?

AFAICS it only happens in one place on `putIssue` – given that the `PublicationTargetBase` trait only has two methods, only one of which implicates an issue, could we not just do the transform locally in the recipe publication target, and save ourselves some polymorphism complexity and this trait?

IIUC (may be missing something) I'd be tempted to wait for another use case before adding a level of indirection here.

Reply:
Yeah maybe I was over-engineering it 🤷